### ISSUE TITLE
(feature request): add shortcut for --prefix

### ISSUE THREAD

**Issue by kuncevic** [npm/cli#6721]
The problem
-

Currently, the `--prefix` option is used to specify a directory that you want to execute npm command for. Typing `--prefix` each time is a lot of work.

Solution
-
I propose a flag `-p` for the npm as a shortcut to quickly specify the `--prefix` option when running various commands.

**Comment by ljharb** [comment]
That arg imo is dangerous and should be avoided, and instead you should cd into the dir you want in a subshell - shortening prefix would encourage its use.

**Comment by kuncevic** [comment]
@ljharb I c your point. However at the same time I believe that `--prefix` is a bit misleading as flag itself. In yarn for example, there is a flag called `--cwd` that act same way, `--cwd` itself self explanatory and short. Having same flag (`--cwd`) in npm could be a great addition, in that case won't need any shortcuts. What do you think?

**Comment by ljharb** [comment]
While I agree that "cwd" is a better name for it than "prefix", I don't think the functionality should exist at all - if you want to change the working directory for a command, change it before running the command.

**Comment by kuncevic** [comment]
I think this functionality is needed, for example in cases where you have multiple projects within git monorepo (non npm/yarn/etc monorepo) 

Being able to execute npm scripts within subdirectories directly from the root folder is incredibly handy; otherwise, you'll end up in a loop of directory acrobatics cd in and cd out of the folders. 

Just for doing that I had to write a custom shell script to solve that problem by typing minimum number characters possible however this is just local workaround that won't work cross-platform or on someone else machine.

**Comment by ljharb** [comment]
@kuncevic it's not acrobatics to do `push` and `popd`, or to `cd` inside a subshell - it's basic terminal stuff.

**Comment by kuncevic** [comment]
Yes `cd` is a basic terminal stuff which is not a question, the question is who want to sit and type `cd` commands all day?

**Comment by kuncevic** [comment]
Yes cd is a basic terminal stuff which is not a question, the question is who want to sit and type cd command all day?

**Comment by kuncevic** [comment]
Yes `cd` is a basic terminal stuff which is not a question, the question is who want to sit and type `cd` command all day?

**Comment by kuncevic** [comment]
Yes `cd` is a basic terminal stuff which is not a question, the question is who want to sit and type `cd` command all day in case that I've referenced above 🤷‍♂️

**Comment by ljharb** [comment]
it's a lot fewer characters than `--prefix`, so i'd think everybody.

**Comment by rotu** [comment]
BTW, it seems `npm exec` also *always* runs in the *actual* working directory, as seen by `npm --prefix=/ exec -c pwd`. This majorly blunts its usefulness.

**Comment by rotu** [comment]
BTW, it seems `npm exec` also *always* runs in the *actual* working directory, as seen by `npm --prefix=/ exec -c pwd`. This majorly blunts the usefulness of `--prefix`.

IMHO, prefix should either be fixed or loudly deprecated. There's no sense in leaving in features that are so broken to use.

**Comment by ljharb** [comment]
I’d hope the latter.

**Comment by kuncevic** [comment]
I've solved that problem that way https://github.com/kuncevic/execute-dir.fish

**Comment by kuncevic** [comment]
I've solved that problem this way https://github.com/kuncevic/execute-dir.fish
